src/backend/distributed/commands/vacuum.c

```diff
@@ -210,20 +212,25 @@ VacuumTaskList(Oid relationId, CitusVacuumParams vacuumParams, List *vacuumColum
 	foreach_ptr(shardInterval, shardIntervalList)
 	{
 		uint64 shardId = shardInterval->shardId;
+		char *shardRelationName = pstrdup(relationName);
+
+		/* build shard relation name */
+		AppendShardIdToName(&shardRelationName, shardId);
 
-		char *shardName = pstrdup(tableName);
-		AppendShardIdToName(&shardName, shardInterval->shardId);
-		shardName = quote_qualified_identifier(schemaName, shardName);
+		char *quotedShardName = quote_qualified_identifier(schemaName, shardRelationName);
 
-		vacuumString->len = vacuumPrefixLen;
-		appendStringInfoString(vacuumString, shardName);
-		appendStringInfoString(vacuumString, columnNames);
+		/* copy base vacuum string and build the shard specific command */
+		StringInfo vacuumAfterString = makeStringInfo();
+		appendStringInfoString(vacuumAfterString, baseVacuumString);
+
+		appendStringInfoString(vacuumAfterString, quotedShardName);
+		appendStringInfoString(vacuumAfterString, columnNames);
 
 		Task *task = CitusMakeNode(Task);
-		task->jobId = jobId;
+		task->jobId = INVALID_JOB_ID;
 		task->taskId = taskId++;
 		task->taskType = VACUUM_ANALYZE_TASK;
-		SetTaskQueryString(task, pstrdup(vacuumString->data));
+		SetTaskQueryString(task, pstrdup(vacuumAfterString->data));
```

Comment:
Do we still need `pstrdup `? We're already allocating a new StringInfo per shard

Reply:
actually no, removing it